src/types/responses.ts

```diff
@@ -7,19 +9,75 @@ export interface ReceivedJSON {
 }
 
 /**
- * Interface for USERS payload received
+ * Interface for LOGIN_SUCCESS payload received
+ */
+export interface LoginSuccessJSON {
+  success: boolean;
+  error?: string;
+}
+
+/**
+ * Interface for GET_ALL_USERS payload received
  */
 export interface UsersJSON {
-  users: Record<string, string>[];
+  users: User[];
+}
+
+/**
+ * Interface for USER_UPDATE payload received
+ */
+export interface UpdateUserJSON {
+  connectionId: string;
+  username: string;
+  state: string;
 }
```

Comment:
so USER_UPDATE means sending the updated info (e.g. state: 'CONNECT') of a user when the user joins a game, but not sending all the user info (User[]) to every user?

and GAME_UPDATE means sending the updated game info (e.g. state: 'CREATED') after a user joins a game, but not sending all the game info (Game[]) to every user?

Reply:
USER_UPDATE means sending updated info of a user when the user is connected to Websocket, but not sending all User[] to every user.

GAME_UPDATE means sending updated game (eg. when a game is CREATED) when a user creates one, or deletes a game, instead of sending Game[] to every user.

A user would only get User[] and Game[] when they land on the lobby page, and would get partial updates through USER_UPDATE and GAME_UPDATE.

Reply:
seems like both `USER_UPDATE` and `GAME_UPDATE` are reactive, I don't need routes for them, right? In other words, do you ever need to call them with these?
```json
{"action": "USER_UPDATE", "payload": {...} }
```
```json
{"action": "GAME_UPDATE", "payload": {...} }
```

Reply:
Yes

Reply:
alright, thx